tests/integration/test_route53resolver.py

```diff
@@ -0,0 +1,570 @@
+            Name=resolver_endpoint_name,
+            IpAddresses=ip_addresses,
+        )
+
+        create_resolver_endpoint_res = result.get("ResolverEndpoint")
+
+        if self._wait_created_endpoint_is_listed_with_status(
+            route53resolver_client, request_id, "OPERATIONAL"
+        ):
+            create_resolver_endpoint_res["Status"] = "OPERATIONAL"
+
+        snapshot.match("create_resolver_endpoint_res", create_resolver_endpoint_res)
+
+        with pytest.raises(Exception) as invalid_request:
+            route53resolver_client.create_resolver_endpoint(
+                CreatorRequestId=request_id,
+                SecurityGroupIds=security_groups_ids,
+                Direction="INBOUND",
+                Name=resolver_endpoint_name,
+                IpAddresses=ip_addresses,
+            )
+        snapshot.match(
+            "invalid_request_error_code", invalid_request.value.response.get("Error").get("Code")
+        )
+        snapshot.match(
+            "invalid_request_error_http_status_code",
+            invalid_request.value.response.get("ResponseMetadata").get("HTTPStatusCode"),
+        )
+
+        cleanups.append(
+            lambda: route53resolver_client.delete_resolver_endpoint(
+                ResolverEndpointId=create_resolver_endpoint_res["Id"]
+            )
+        )
+
+    @pytest.mark.aws_validated
+    @pytest.mark.skip_snapshot_verify(paths=["$..SecurityGroupIds"])
+    def test_update_resolver_endpoint(self, ec2_client, route53resolver_client, cleanups, snapshot):
+        request_id = short_uid()
+        result = route53resolver_client.create_resolver_endpoint(
+            CreatorRequestId=request_id,
+            SecurityGroupIds=self.get_security_group_ids(ec2_client),
+            Direction="INBOUND",
+            IpAddresses=self.get_create_resolver_endpoint_ip_address(ec2_client, "58"),
+        )
+
+        create_resolver_endpoint_res = result.get("ResolverEndpoint")
+
+        if self._wait_created_endpoint_is_listed_with_status(
+            route53resolver_client, request_id, "OPERATIONAL"
+        ):
+            create_resolver_endpoint_res["Status"] = "OPERATIONAL"
+        snapshot.match("create_resolver_endpoint_res", create_resolver_endpoint_res)
+
+        # update resolver endpoint
+        update_resolver_endpoint_res = route53resolver_client.update_resolver_endpoint(
+            ResolverEndpointId=create_resolver_endpoint_res["Id"], Name="resolver_endpoint_name"
+        )
+
+        if self._wait_created_endpoint_is_listed_with_status(
+            route53resolver_client, request_id, "OPERATIONAL"
+        ):
+            update_resolver_endpoint_res["Status"] = "OPERATIONAL"
+        snapshot.match("update_resolver_endpoint_res", update_resolver_endpoint_res)
+
+        cleanups.append(
+            lambda: route53resolver_client.delete_resolver_endpoint(
+                ResolverEndpointId=create_resolver_endpoint_res["Id"]
+            )
+        )
```

Comment:
see other cleanup remarks

Reply:
Some cleanups are not possible just after the creation as they are dependent, eg: if I cleanup endpoint before rules, it throws error.

Reply:
same with the associations

Reply:
@dominikschubert there are some dependent relations eg: we cant delete a resolver endpoint until all its rules and for rules all the associations are deleted, but will cleanups ensure this behaviour? (The reason why it was added it at the end of the all the creations)

Reply:
What you add to the cleanups will generally be executed in reverse order after the test execution. So if it's the last you want to be deleted, it should be the first added to the cleanups.

Reply:
got it